bip-0341.mediawiki

```diff
@@ -141,6 +141,12 @@ To validate a signature ''sig'' with public key ''q'':
 * If the ''sig'' is 65 bytes long, return ''sig[64] &ne; 0x00<ref>'''Why can the <code>hash_type</code> not be <code>0x00</code> in 65-byte signatures?''' Permitting that would enable malleating (by third parties, including miners) 64-byte signatures into 65-byte ones, resulting in a different `wtxid` and a different fee rate than the creator intended</ref> and Verify(q, hash<sub>TapSighash</sub>(0x00 || SigMsg(sig[64], 0)), sig[0:64])''.
 * Otherwise, fail<ref>'''Why permit two signature lengths?''' By making the most common type of <code>hash_type</code> implicit, a byte can often be saved.</ref>.
 
+==== Taproot script path spending signature validation ====
+
+Validation is identical to ''key path validation'' except that the input to hash<sub>TapSighash</sub> is ''(0x00 || SigMsg(hash_type, ext_flag) || ext)''.
+
+The value of ''ext'' is computed as specified in the BIP corresponding a particular value of ''ext_flag''.
+
```

Comment:
That's a semantic change of the BIP and precludes new leaf versions from computing the signature message hash differently. There's no reason to do that.

Also the beginning of the section `Signature validation rules` mentions that it only applies to key spending.

Reply:
Mmm. Right, will remove.